redditnav.js

```diff
@@ -1,145 +1,102 @@
-// Copyright (c) 2014 The Chromium Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-UP = 0
-DOWN = 1
+var directions = {
+  UP: "up",
+  DOWN: "down"
+};
+var scrolling = false;
 
-function goToNextParent(pos, direction) {
-  var parentComments = $(".sitetable.nestedlisting").children(".comment").toArray();
-  parentComments = parentComments.map(function(commentElement){
-    return $(commentElement);
-  });
+// http://gizma.com/easing/#quad3
+function easeInOutQuad(n,u,e,t){return n/=t/2,1>n?n*n*(e/2)+u:(--n,(n*(n-2)-1)*(-e/2)+u)}
 
-  var $scrollTo = getNextParent(Math.ceil(pos), direction, parentComments);
-  if($scrollTo == null){
+function animateScrollTo(position, duration) {
+  if (scrolling)
     return;
+
+  var start = null;
+  var scrollY = window.scrollY;
+  function step(timestamp) {
+    scrolling = true;
+    if (!start)
+      start = timestamp;
+
+    var progress = timestamp - start;
+    var top = easeInOutQuad(progress, scrollY, position - scrollY, duration);
+    window.scroll(0, top);
+    if (progress < duration)
+      window.requestAnimationFrame(step);
+    else
+      scrolling = false;
   }
-  console.log("to: " + getPos($scrollTo) + ", from: " + Math.ceil(pos)  + ", " + direction)
+  window.requestAnimationFrame(step);
+}
 
-  $("body").animate({
-      scrollTop: getPos($scrollTo)
-  });
-  $scrollTo.children(".entry").click();
+function getPos(node) {
+  return node.getBoundingClientRect().top + document.body.scrollTop;
 }
 
 function getNextParent(pos, direction, parentComments) {
-  if(pos < getPos(parentComments[0])){
-    if(direction == DOWN)
-      return parentComments[0];
-    else
-      return null;
-  }
-
-  if(pos > getPos(parentComments[parentComments.length-1])){
-    if(direction == UP)
-      return parentComments[parentComments.length-1];
-    else
-      return null;
+  var currentIndex = 0;
+  for (var i = 0; i < parentComments.length; ++i) {
+    var parentPos = getPos(parentComments[i]);
+    if (pos > parentPos || (direction === directions.DOWN && Math.abs(pos - parentPos) < 5))
```

Comment:
I think it may be a resolution thing? What's the resolution on the screen you're using @dcrousso ? I remember having this issue come up while we were first starting the project, but I can't really remember what we did to fix it exactly.

See video of it here: http://goo.gl/NklgDS
It's kind of hard to tell, but at around 11-13 seconds when I reach the Top Gear comment it'll scroll up a couple pixels on pressing q as opposed to going all the way up to the next comment. Doesn't seem to occur while going down at all anymore though.

Also, found a funny bug while I was playing with the floating button: http://goo.gl/v7R5jK